app/ante/reject_msgs.go

```diff
@@ -9,28 +9,43 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/authz"
 	govtypesv1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
 	"github.com/cosmos/cosmos-sdk/x/group"
+	"github.com/dymensionxyz/gerr-cosmos/gerrc"
 	evmtypes "github.com/evmos/ethermint/x/evm/types"
 )
 
 // RejectMessagesDecorator prevents invalid msg types from being executed
 type RejectMessagesDecorator struct {
-	disabledMsgTypeURLs map[string]struct{}
+	// message is rejected if any Predicate returns true
+	predicates []Predicate
 }
 
-var _ sdk.AnteDecorator = RejectMessagesDecorator{}
+// Predicate should return true if message is not allowed
+type Predicate = func(typeURL string, depth int) bool
```

Comment:
```suggestion
// predicate should return true if message is not allowed
type predicate = func(typeURL string, depth int) bool
```

Reply:
Can't do it because of crappy cosmos sdk pattern to have every test in _test package